# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/Texture.cpp
# Commit: 03d61eb955a9
# Full Hash: 03d61eb955a9ac6e3374ca4c5504edc41cd1aa4a
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2023-10-25 16:28:20
# Description:
#   Bug 1860921 - Revert Texture::Destroy() and add Texture::ForceDestroy() r=webgpu-reviewers,nical
#   
#   This is a short term fix. Long term fix needs to be done by Bug 1860958.
#   Presenting WebGPU without readback needs a way to destroy Texture. Then Texture::ForceDestroy() is added for it.
#   
# ==============================================================================

diff -r dc00edd0eac6 -r 03d61eb955a9 dom/webgpu/Texture.cpp
--- a/dom/webgpu/Texture.cpp	Wed Oct 25 09:41:31 2023 +0000
+++ b/dom/webgpu/Texture.cpp	Wed Oct 25 09:43:58 2023 +0000
@@ -67,6 +67,13 @@
   return view.forget();
 }
 
-void Texture::Destroy() { Cleanup(); }
+void Texture::Destroy() {
+  // TODO: we don't have to implement it right now, but it's used by the
+  // examples
+
+  // XXX Bug 1860958.
+}
+
+void Texture::ForceDestroy() { Cleanup(); }
 
 }  // namespace mozilla::webgpu